lib/rubycritic/source_control_systems/git.rb

```diff
@@ -52,6 +52,18 @@ def stashes_count
       def travel_to_original_state
         `git stash pop`
       end
+
+      def path_revisions_counts
+        `git log --name-only --format='' | sort | uniq -c`
```

Comment:
@guilhermesimoes Interesting! I'll try and check to see what that does on a repo that's under the limit. It's not clear to me if that would also cause a slowdown on a small repo.

@Onumis I think you hit the nail on the head. Someone could have renamed a file from a directory they ignore now into a directory they care about. Might not be that common of an occurrence though. I think it might be best to play it safe and look at everything. What do you think?